File: client/src/lib/solana/model/TransactionState.ts
import { add_enum_to_schema, Enum, UnitValue } from '../solanaBorsh';

export class TransactionState extends Enum<TransactionState> {
  notReady?: UnitValue;
  ready?: UnitValue;
  executed?: UnitValue;

  constructor(
    props:
      | { notReady: UnitValue }
      | { ready: UnitValue }
      | { executed: UnitValue }
  ) {
    super(props);
  }
}

add_enum_to_schema(TransactionState, {
  notReady: UnitValue,
  ready: UnitValue,
  executed: UnitValue,
});

File: programs/cryptid_anchor/src/state/transaction_state.rs
use anchor_lang::prelude::*;

/// A [`TransactionAccount`]'s state
#[derive(Clone, Debug, AnchorDeserialize, AnchorSerialize, PartialEq)]
pub enum TransactionState {
    /// Transaction account is not ready to execute - it is waiting for additional instructions to be added
    NotReady,
    /// Transaction account is ready to execute
    Ready,
    /// Transaction account has executed
    Executed,
}
impl TransactionState {
    /// Calculates the on-chain size of a [`TransactionState`]
    pub const fn calculate_size() -> usize {
        1 //enum
    }
}
impl Default for TransactionState {
    fn default() -> Self {
        Self::NotReady
    }
}
